defaults/main.yml

```diff
@@ -1,7 +1,32 @@
+# note: the booleans are quoted otherwise they are rendered capitalised.
 ---
+buildkite_agent_major_version: 3
 buildkite_agent_conf_dir: "/etc/buildkite-agent"
 buildkite_agent_token: ""
-buildkite_agent_name: "buildkite-agent"
+buildkite_agent_debug: "false"
+buildkite_agent_enabled: true
+# if you're building a golden image and don't want it to connect to
+# buildkite during build ptocess, set this to 'stopped'.
+buildkite_agent_state: "started"
+
+# v2 https://buildkite.com/docs/agent/v2/configuration
```

Comment:
Just as a heads up that I'd be OK with breaking backwards compatibility with the v2 agent. We'd likely just bump the major version of the Ansible role to help communicate that.

Reply:
So, I shouldn't bother with the v2 template at all?

Reply:
Na, feel free to focus on only v3 support.

Reply:
Were you going to remove the v2 agent stuff out in this PR?